### ISSUE TITLE
`Module "util" has been externalized for browser compatibility` when using @comunica/query-sparql with Vite

### ISSUE THREAD

**Issue by renyuneyun** [comunica/comunica#1136]
#### Issue type:

- :bug: Bug 

____
#### Description:

I'm developing [a Solid App](https://github.com/renyuneyun/PermiX), which uses Comunica for some queries. It uses Vite for building. I assume this should work by default ([Comunica's document](https://comunica.dev/docs/query/getting_started/query_browser_app/)), as Vite should be a web bundling tool based on my understanding.

Under dev build mode, it works, and I can get my query results.
However, I also get the following (warning) messages in the browser console:

```
Module "util" has been externalized for browser compatibility. Cannot access "util.debuglog" in client code. [null:24575:19](http://localhost:5173/node_modules/.vite/deps/null)
Module "util" has been externalized for browser compatibility. Cannot access "util.inspect" in client code. [null:24575:19](http://localhost:5173/node_modules/.vite/deps/null)
```

In production mode, it won't work. I'm still trying to find out why, but guess this might be one of the reasons?
____
#### Environment:

| software         | version
| ---------------- | -------
| Comunica Engine  | 2.5.1
| node             | v19.0.1
| npm              | 8.19.2
| yarn             | _Yarn is unavailable_
| Operating System | darwin (Darwin 22.1.0)

**Comment by rubensworks** [comment]
Never used Vite myself, so not sure why that could be failing.
Comunica shouldn't be using `util` directly AFAIK.

@surilindur If I remember correctly, you used Comunica with Vite, right?

**Comment by surilindur** [comment]
Sorry about the delay, I did not have the time to look into this. The warnings about `util` seem to be related to version `3.6.0` of `readable-stream` that imports `util` and is used by `readable-web-to-node-stream`. Also `winston` and `winston-transport` have that version but they do not seem to produce any warnings. Adding a resolve alias to Vite config that points at nothing seems to indicate that:
```
  resolve: {
    alias: {
      util: 'nothing'
    }
  }
```
```
node_modules/readable-web-to-node-stream/node_modules/readable-stream/lib/_stream_readable.js:60:24:
  60 │ var debugUtil = require('util');
```
The import as it was in `3.6.0` is not there anymore in the `4.6.0` version of `readable-stream` that gets installed in the node modules and is used by everything else, it seems.

I managed to get rid of the warning by making an alias for `util` that points at the internal `util` library from the new version of `readable-stream`:
```
  resolve: {
    alias: {
      util: 'node_modules/readable-stream/lib/ours/util.js'
    }
  }
```
It looks weird and 100% likely to break at some point but gets rid of the warning for me, at least.

I also ran into the same issue with `undefined is not a non-null object` or a similar error with a production build using Vite (Vue + Comunica), that only happened in production mode. I will try to look into it more. I changed to Webpack as a result to get the build working, but I really want to use Vite instead.

**Comment by rubensworks** [comment]
> The import as it was in 3.6.0 is not there anymore in the 4.6.0 version of readable-stream

So if I understand it correctly, if we just bump that one to the latest version, things will be fixed?
If so, can you look into making a PR to Comunica for this?

**Comment by surilindur** [comment]
I think the version might need to be bumped in `readable-web-to-node-stream` (at least), and it will only solve the warning about `util`, since Comunica uses `4.6.0` but that library has its own `readable-stream` that is older. I will try to do it locally and see if it is enough.

**Comment by rubensworks** [comment]
Ah ok. Worst-case we can fork `readable-web-to-node-stream` and republish it with a bumped version (or we can pull in the code of `readable-web-to-node-stream` into Comunica, so we can remove the dependency).

**Comment by rubensworks** [comment]
@surilindur Any news on this issue?

**Comment by surilindur** [comment]
Not yet, sorry, I have not had the time to look into it. I will try to do it later this week.

**Comment by rubensworks** [comment]
Any news @surilindur?

**Comment by surilindur** [comment]
Sorry about taking so long again. I tried Comunica with Vite again (totally forgot what I did previous time) and the warning only seems to appear with the development server mode of Vite, and is gone in the production build. It also does not seem to have any impact on the functionality.

There is a small application that uses Comunica to fetch any triples it finds [here](https://github.com/surilindur/minimal-vite-setup), that I used to test if Vite works. If I have time, I will try to run the application this issue was created for to see what happens. Maybe something else relies on the `util` module directly and breaks as a result, or the application that breaks is using Comunica in a different way?
